src/rtcModule/webrtc.cpp

```diff
@@ -940,82 +971,58 @@ void RtcModule::abortCallRetry(Id chatid)
             }
 
-            size_t posInitialColon = fullUrl.find(":") + 1;
-            size_t posFinalColon = fullUrl.rfind(":");
-            std::string url = fullUrl.substr(posInitialColon, fullUrl.size() - posInitialColon);
-            std::string host = fullUrl.substr(posInitialColon, posFinalColon - posInitialColon);
-            auto wptr = weakHandle();
-            DnsRequest::getInstance()->wsResolveDNS(mKarereClient.websocketIO, host.c_str(),
-                                                    [wptr, this, url, shard, dnsRequestId]
-                                                    (int statusDNS, const std::vector<std::string> &ipsv4, const std::vector<std::string> &ipsv6)
+            if (statusDNS < 0)
             {
-                if (wptr.deleted() || dnsRequestId != mDnsRequestId)
-                    return;
-
-                if (mKarereClient.isTerminated())
-                {
-                    RTCM_LOG_ERROR("DNS resolution completed but karere client was terminated.");
-                    return;
-                }
-
-                if (statusDNS < 0)
-                {
-                    RTCM_LOG_ERROR("Async DNS error in rtcModule. Error code: %d", statusDNS);
-                }
+                RTCM_LOG_ERROR("Async DNS error in rtcModule. Error code: %d", statusDNS);
+            }
 
-                if (!mKarereClient.mDnsCache.hasRecord(shard))
-                {
-                    mKarereClient.mDnsCache.addRecord(shard, url);
-                }
-                else if (mKarereClient.mDnsCache.getUrl(shard).host != url)
-                {
-                    assert(false); // They should be removed at begining of the method
-                    mKarereClient.mDnsCache.removeRecord(shard);
-                    mKarereClient.mDnsCache.addRecord(shard, url);
-                }
+            assert(mKarereClient.mDnsCache.hasRecord(shard));
 
-                if (statusDNS >= 0 && (ipsv4.size() || ipsv6.size()))
-                {
-                    RTCM_LOG_ERROR("New IP for TURN servers: ipv4 - %s     ipv6 - %s",
-                                 ipsv4.size() ? ipsv4[0].c_str() : "",
-                                 ipsv6.size() ? ipsv6[0].c_str() : "");
-                    mKarereClient.mDnsCache.setIp(shard, ipsv4, ipsv6);
-                }
+            if (statusDNS >= 0 && (ipsv4.size() || ipsv6.size()))
+            {
+                RTCM_LOG_ERROR("New IP for TURN servers: ipv4 - %s     ipv6 - %s",
+                             ipsv4.size() ? ipsv4[0].c_str() : "",
+                             ipsv6.size() ? ipsv6[0].c_str() : "");
+                mKarereClient.mDnsCache.setIp(shard, ipsv4, ipsv6);
+            }
 
-                StaticProvider iceServer;
-                string jsonCacheTurnServer = getCachedTurnServers();
-                iceServer.setServer(jsonCacheTurnServer.c_str());
-                setIceServers(iceServer);
-            });
+            updateTurnServer();
```

Comment:
Do we actually update anything in the TURN servers passed to WebRTC when the `wsResolveDNS()` fails?

Reply:
Fixed it